src/charm.py

```diff
@@ -413,6 +422,9 @@ def _prometheus_layer(self) -> Layer:
                     "summary": "prometheus daemon",
                     "command": self._generate_command(),
                     "startup": "enabled",
+                    "environment": {
+                        "OTEL_RESOURCE_ATTRIBUTES": f"juju_application={self._topology.application},juju_model={self._topology.model},juju_model_uuid={self._topology.model_uuid},juju_unit={self._topology.unit}"
```

Comment:
I think we also need `juju_charm`?

Reply:
I think I didn't add it because it was already present as service.name, it felt odd to have duplicate data. but we should have it in the workload tracing.

Reply:
also I'm open to add it to charm-tracing-traces too if you think it's better

Reply:
Should be all good now.